### PR TITLE
Add support for kubernetes auth

### PR DESCRIPTION
**Pull request by UXabre**
This provides support for authenticating using kubernetes as discussed here:
https://github.com/fffonion/lua-resty-acme/issues/36

It adds the options to:
 - Set auth_method (either token or kubernetes)
 - Set jwt_path (where the service account token can be found)
 - Set the role to associate with the token

### CODE REVIEW

File: lib/resty/acme/storage/vault.lua [hunk 1]
```diff
@@ -47,16 +52,21 @@ function _M.new(conf)
     mt
   )
 
-  self.headers = {
-    ["X-Vault-Token"] = conf.token,
-  }
-  if self.https then
-    if not self.tls_server_name then
-      self.tls_server_name = self.host
+  local token, err = auth(self, conf)
```
[the remark this answers is not in the document]
Comment (UXabre) [reply lib/resty/acme/storage/vault.lua:55]: I'm not sure I understand? Is it not caught in the next part where I only continue if there is no error? (Line 57)
I just noticed I did not return this error though... I'll fix that

[the remark this answers is not in the document]
Comment (fffonion) [reply lib/resty/acme/storage/vault.lua:55]: we will also need to popup the error to caller, otherwise on error `new()` will return
`nil, nil`

**Comment by fffonion** [comment]
Thanks a lot @UXabre ! Would you mind update the Vault section in Readme and also add yourself to Credits?

**Comment by UXabre** [comment]
I'll add myself to the credits and see to add more info to the readme regarding this adjustment (I only added the variables but I'll see where I can elaborate)! Thanks for the great teamwork!

**Pull request merged**